examples/js/loaders/GLTFLoader.js

```diff
@@ -2772,126 +2831,148 @@ THREE.GLTFLoader = ( function () {
 
 		var animationDef = json.animations[ animationIndex ];
 
-		return this.getMultiDependencies( [
+		var nodePendings = [];
+		var inputAccessorPendings = [];
+		var outputAccessorPendings = [];
+		var samplerPendings = [];
+		var targetPendings = [];
 
-			'accessor',
-			'node'
+		for ( var i = 0, il = animationDef.channels.length; i < il; i ++ ) {
 
-		] ).then( function ( dependencies ) {
+			var channel = animationDef.channels[ i ];
+			var sampler = animationDef.samplers[ channel.sampler ];
 
-			var tracks = [];
+			if ( sampler === undefined ) continue;
```

Comment:
Thanks, removed. BTW off topic and nit, which one does that schema indicate?
- only `channel.sampler` is required
- `channel.sampler` is required and also `animationDef[ channel.sampler ]` is required

In either case, perhaps validator checks `animationDef[ channel.sampler ]` is not undefined, so agreed removing.

Reply:
I think you mean `channel.sampler` or `animationDef.samplers[ channel.sampler ]` right? Both are required, so it would be reasonable for a runtime error to happen if either were missing.

On a related note, `target.node` is _not_ required (https://github.com/KhronosGroup/glTF-Asset-Generator/issues/453), but let's fix that in another PR.

Reply:
I meant, I know my concern is very nit tho, `channel.sampler` is required

https://github.com/KhronosGroup/glTF/blob/master/specification/2.0/schema/animation.channel.schema.json#L20

and also `animation.samplers` is required

https://github.com/KhronosGroup/glTF/blob/master/specification/2.0/schema/animation.schema.json#L28

but I couldn't find `animations.samplers[ channel.sampler ]` is required anywhere. Does this line indicate so? 

https://github.com/KhronosGroup/glTF/blob/master/specification/2.0/schema/animation.channel.schema.json#L9

(Should it be `"allOf": [ { "$ref": "animation.sampler.schema.json" } ],` by any chance?)

Reply:
`channel.sampler` must be provided, and must be the index of a sampler in `animation.samplers`. If it's just provided with some random value like `-1`, or `10` when the length of `animation.samplers` is 5, it is still invalid.

Reply:
> must be the index of a sampler in animation.samplers. If it's just provided with some random value like -1, or 10 when the length of animation.samplers is 5, it is still invalid.

My question is, where is it specified in the spec/schema? It isn't explicitly specified but worded somewhere in the spec?

Reply:
That's the definition of `channel.sampler` — 

> A channel's sampler property contains the index of one of the samplers present in the containing animation's samplers array.

Reply:
OK, I've missed that.